### PR TITLE
add firebase cloud function for API

### PR DESCRIPTION
**Pull request by fadhilmch**
Add some APIs to get, add, delete, and update books data using Firebase Cloud Function.
Add getBooksByDistance API to collect data based on the nearest location from user position. (Compare it one by one, still not efficient)

### CODE REVIEW

File: functions/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (bt-s) [inline functions/.gitignore]: Do we need separate .gitignore and .eslintrc files for the Firebase functions folder?

If it also works with our global .gitignore and .eslintrc files, I propose that we delete these local ones.

File: functions/index.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+const cors = require('cors')({origin: true});
+const geolib = require('geolib');
+
+admin.initializeApp();
+
+const database = admin.firestore().collection('books');
+// // Create and Deploy Your First Cloud Functions
```
Comment (bt-s) [inline functions/index.js]: The next 5 lines can be deleted.

**Review by bt-s — COMMENTED** [review]
Looks good, Fadhil!

In this version, I am not able to upload an image on the Add book page.
Could you please investigate why this does not work?
If this does not work because the ```firebase-cloud-func``` branch was branched of from an older version of ```master```, feel free to merge it!

**Comment by fadhilmch** [comment]
this branch was created before the add_book is merged, that is why it is still not working on this branch

**Review by bt-s — APPROVED** [review]

**Pull request merged**